examples/pipelines/commoncrawl/components/download_commoncrawl_segments/src/utils/download_utils.py

```diff
@@ -0,0 +1,42 @@
+import time
+import logging
+
+import boto3
+
+import requests
+from requests import RequestException, ConnectionError
+
+logger = logging.getLogger(__name__)
+
+S3_COMMONCRAWL_BUCKET = "commoncrawl"
+COMMONCRAWL_BASE_URL = "https://data.commoncrawl.org/"
+
+
+def get_warc_file_using_boto3(s3_key: str) -> bytes:
+    """Downloads a WARC file using boto3.
+    Args:
+        warc_file: The path to the WARC file.
+    Returns:
+        The WARC file as a bytes object.
+    """
+    s3 = boto3.client("s3")
+    response = s3.get_object(Bucket=S3_COMMONCRAWL_BUCKET, Key=s3_key)
+    return response["Body"]
+
+
+def get_warc_file_using_requests(warc_file: str) -> requests.Response:
+    retry = 0
```

Comment:
FYI, requests provides functionality for retry:
https://requests.readthedocs.io/en/latest/user/advanced/?highlight=retry#example-automatic-retries